rdt/hyper_transformer.py

```diff
@@ -442,21 +442,17 @@ def update_transformers(self, column_name_to_transformer):
         self._validate_update_columns(update_columns)
         self._validate_transformers(column_name_to_transformer)
 
-        incompatible_sdtypes = []
         for column_name, transformer in column_name_to_transformer.items():
             if transformer is not None:
                 current_sdtype = self.field_sdtypes.get(column_name)
                 if current_sdtype and current_sdtype not in transformer.get_supported_sdtypes():
-                    incompatible_sdtypes.append(column_name)
+                    raise InvalidSdtypeForTransformerError(
+                        f"Column '{column_name}' is a {current_sdtype} column, which is "
+                        f"incompatible with the '{type(transformer).__name__}' transformer."
```

Comment:
🤔 Should we add a `get_name` method to the `BaseTransformer` and use that whenever we want to get the `__name__` ?